firstvoices/backend/resources/utils/import_export_widgets.py

```diff
@@ -0,0 +1,44 @@
+from import_export.widgets import ForeignKeyWidget, Widget
+
+from backend.models.user import User
+
+
+class ChoicesWidget(Widget):
+    """Import/export widget to use choice labels instead of internal value.
+
+    Maps display labels to db values for import, and vvsa for export.
+    """
+
+    def __init__(self, choices, *args, **kwargs):
+        """
+        Input:
+        - choices: iterable of choices containing (dbvalue, label)
+            e.g. [(20, "Public"), ...]
+        """
+        self.choice_labels = dict(choices)
+        self.choice_values = {v: k for k, v in choices}
+
+    def clean(self, value, row=None, *args, **kwargs):
+        """Returns the db value given the display value"""
+        return self.choice_values.get(value) if value else None
+
+    def render(self, value, obj=None):
+        """Returns the display value given the db value"""
+        return self.choice_labels.get(value)
+
+
+class UserForeignKeyWidget(ForeignKeyWidget):
+    """Import/export widget to find users by their email, and create in django if missing."""
+
+    def __init__(self, create=False, *args, **kwargs):
+        self.create = create
+        super().__init__(model=User, field="email", *args, **kwargs)
+
+    def clean(self, value, row=None, **kwargs):
+        if self.create:
+            user, _ = self.model.objects.get_or_create(
+                **{self.field: value}, defaults={"id": value}
+            )
```

Comment:
It looks like this is creating users without a password, meaning the password will be NULL in the database. I think this is fine for now and we can update this in the future when we have a way for users to claim migrated accounts, but we should probably make a note of this in one of the user management epic tickets. 

Additionally, the join date on the user model is getting set to the import time and we may want to consider migrating a join date if that exists in the data.

Reply:
Definitely agreed. I will create-or-update a jira ticket about migration of user accounts and metadata to django specifically. We'll also need to think about collisions and fixes for the non-case-sensitive emails, which I didn't handle here.